File: Back/src/main/java/com/gestion/adhesion/models/Paiement.java
package com.gestion.adhesion.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@EqualsAndHashCode(of = {"id"})
@Entity
@Table(	name = "paiements",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = "id"),
        })
@JsonIgnoreProperties(ignoreUnknown = true)
public class Paiement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate dateReglement;

    private String typeReglement;

    private Integer montant;

}

File: client/src/app/models/paiement.ts
export class Paiement {

  id: number;
  montant: number;
  typeReglement: String;
  dateReglement: String;

  constructor(){
    this.id = 0;
    this.montant = 0;
    this.typeReglement = "";
    this.dateReglement = "";
  }
}
